src/transformers/models/albert/modeling_albert.py

```diff
@@ -1150,16 +1150,7 @@ def forward(
         loss = None
         if labels is not None:
             loss_fct = CrossEntropyLoss()
-            # Only keep active parts of the loss
```

Comment:
Instead of breaking it, could we maybe deprecate it? IMO, there are still quite some people that train models without using our `Trainer`, our `DataCollators`, or official examples. In such cases, I can see people passing labels to the tokenizer:

```
tokenizer(label_str, return_tensors="pt", padding=True)
```

=> in this case the labels would be padded with `0's` which if we remove this part won't be ignored anymore. So IMO it would be nice to go through a deprecation cycle here

Reply:
Sure, I will add deprecation.

Reply:
I don't understand what you're saying @patrickvonplaten . For token classification tasks, labels can't go through the tokenizer like this without making any sense to the model.

Reply:
Like this would encode labels `["O", "B-PER", "I-PER"]` in something completely different than the expected `[0, 1, 2]` and putting label indices as zeros mean the "O" class, not ignore.

Reply:
(I guess I somehow mixed the general token classification tasks with the language modeling task here ...)

Reply:
I also have a hard time to add deprecation - adding this means we want to allow PT models to use attention_mask (for now), but the enhanced test (pt/tf equivalence) will fail, because TF models don't use `attention_mask` (and the labels used for testing don't have `-100` - unless we change `def _prepare_for_class` in `test_modeling_tf_common.py`)

Reply:
Just took a look into the `run_ner.py`. Yes sorry, I didn't thought carefully enough - the use case I'm referring only really applies to the LM case.

Okey good for me then!